### PR TITLE
Anya/4636 endpoint to create omo request

### PR DESCRIPTION
**Pull request by aroltsch**
connects #4636 

TODO: 
In the next PR, I will re-factor the code to use DECASS uniq ID instead of the attorney's CSS id and Vacols ID. 

Example post request:

```
curl -X POST \
  http://localhost:3000/queue/tasks/3076166/complete \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/json' \
  -d '{ 
	"type": "OMORequest",
	"reviewing_judge_id": 3,
	"work_product": "OMO - IME",
	"document_id": "123456789.1234",
	"overtime": true,
	"note": "something"
}'
```

Validate through rails console:
```
AttorneyCaseReview.complete!(vacols_id: "123456", attorney_id: 2, reviewing_judge_id: 3, type: "OMORequest", document_id: "123456789.1234", work_product: "OMO - IME", overtime: false) 
```

### CODE REVIEW

File: app/repositories/queue_repository.rb [hunk 1]
```diff
@@ -36,46 +37,51 @@ def self.appeals_from_tasks(tasks)
...
   #  work_product: "OMO - IME",
   #  overtime: true,
   #  document_id: "123456789.1234",
   #  note: "Require action"
   # }
-  def self.reassign_case_to_judge(decision_record:, decision_hash:)
-    return unless decision_record
-
-    VacolsHelper.validate_presence(decision_hash, [:work_product, :document_id, :judge_css_id])
+  # TODO: use decass uniq ID instead of vacols_id and attorney_css_id
```
Comment (aroltsch) [inline app/repositories/queue_repository.rb]: I will take care of it in the next PR.

File: app/repositories/queue_repository.rb [hunk 2]
```diff
@@ -36,46 +37,51 @@ def self.appeals_from_tasks(tasks)
...
+  def self.reassign_case_to_judge(decass_hash)
+    decass_record = find_decass_record(decass_hash[:vacols_id], decass_hash[:attorney_css_id])
+    fail ReassignCaseToJudgeError unless decass_record
```
Comment (aroltsch) [inline app/repositories/queue_repository.rb]: If decass_record was not found for any reason, we need to fail.

File: app/controllers/queue_controller.rb [hunk 3]
```diff
@@ -19,6 +19,12 @@ def index
     render "queue/index"
   end
 
+  def complete
```
Comment (amprokop) [inline app/controllers/queue_controller.rb]: Eventually, we'll incorporate lots of types of tasks, and we'll likely have to move all the logic from here out somewhere. We might later decide most of the logic here would be better suited for the AttorneyCaseReview class, which could have a `validate` method, etc. But this is fine for now.

However,

File: app/models/attorney_case_reviews/attorney_case_review.rb [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
+          repository.reassign_case_to_judge(
+            vacols_id: vacols_id,
+            attorney_css_id: record.attorney.css_id,
+            judge_css_id: record.reviewing_judge.css_id,
+            work_product: record.work_product,
+            document_id: record.document_id,
+            overtime: record.overtime,
+            note: record.note
+          )
+        rescue *EXCEPTIONS
```
Comment (amprokop) [inline app/models/attorney_case_reviews/attorney_case_review.rb]: interesting approach. This  `ActiveRecord::Rollback` error causes the Caseflow DB update to roll back, right?

File: app/repositories/queue_repository.rb (hunk 2 above)
Comment (amprokop) [reply app/repositories/queue_repository.rb]: 👍

File: app/models/attorney_case_reviews/attorney_case_review.rb (hunk 4 above)
Comment (aroltsch) [reply app/models/attorney_case_reviews/attorney_case_review.rb]: Yes!

File: app/repositories/queue_repository.rb (hunk 2 above)
Comment (amprokop) [reply app/repositories/queue_repository.rb]: This case is essentially our protection against users creating OMO requests for each other as well.

**Pull request merged**